internal/controller/resource_manager.go

```diff
@@ -25,8 +26,8 @@ type ResourceManager struct {
 }
 
 // DynamicListAndWatchResources lists and watches the specified resource dynamically, managing context cancellation and synchronization with wait groups.
-func (r *ResourceManager) DyanmicListAndWatchResources(ctx context.Context, cancel context.CancelFunc, resource string, apiGroup string, allResourcesSnapshotted *sync.WaitGroup, snapshotCompleted *sync.WaitGroup) {
-	resourceListVersion, err := r.DynamicListResources(ctx, resource, apiGroup)
+func (r *ResourceManager) DyanmicListAndWatchResources(ctx context.Context, cancel context.CancelFunc, resource string, apiGroup string, clientset *kubernetes.Clientset, allResourcesSnapshotted *sync.WaitGroup, snapshotCompleted *sync.WaitGroup) {
```

Comment:
This `ClientSet` should be a field of `ResourceManager` instead of new parameters to its methods.